jsonrpc/eth_endpoint.go

```diff
@@ -167,6 +170,8 @@ func (e *Eth) SendRawTransaction(buf argBytes) (interface{}, error) {
 		return nil, err
 	}
 
+	tx = fillTxFees(tx, e.store.GetBaseFee())
```

Comment:
You cannot change the fields from a signed transaction sent by `JsonRPC`. The user signed this transactions, the fields cannot be mutated.

Reply:
We don't really mutate it, just populate either gas price OR gas fee cap + gas tip cap. This is used within the node only and gets broadcasted properly. Gas price is used in many places in the code so we should define it if not provided.

Reply:
Also, this part is modified in the next PR in the chain: https://github.com/0xPolygon/polygon-edge/pull/1230

Reply:
Well, you are mutating the struct if you change the value of the fields. I think you are not mutating it enough to make it change the hash since the `rlp` function will discard the fields depending on the type. However, this changes an imporant invariant of the `Transaction`, this is, the idea that `Transaction` is an immutable object that has a lifecycle that does not change. Instead, now the `Transaction` is a wrapper around a user action.

Reply:
Well, it would require a lot of refactoring which is overkill imo. Another option is to calculate gas price every time when we use it but this solution is completely inefficient. Also, We could delegate this part to the user but it might be really confusing. 
From my side, I don't see potential issues with this. We mutate the tx object when calculating hash as well. 
Do you have any suggestions?

Reply:
Well, if we were to populate fields from the `Transaction`, we cannot do it here only. We should also do it over the p2p and the storage layer.
I do not have the full scope of where the gas price is used. The solution would depend on that. My mental model tells me that the "gas price" (or any other type of sorting function) is required over the txpool mainly. In that case, a possible solution is to create a `txpool.Transaction` wrapper over the `Transaction` which decides how is the sorting factor calculated.

Reply:
It is used in the following places: 
1. Tx pool
2. State Executor
3. Blockchain
4. JSON RPC
5. Tx signer
6. Types
7. Several tests

> Well, if we were to populate fields from the Transaction, we cannot do it here only. We should also do it over the p2p and the storage layer.

I could update RLP marshaling for storage so it would store all these 3 fields regardless of the tx type. The next PR contains some changes related to this part so it does this in the following places: 
1. JSON RPC when receiving transactions.
2. Entrypoint of tx pool for all incoming transactions including ones coming from the network (gossip). 
3. When decoding a transaction while reading it via RPC request.

Reply:
Actually, I agree with you that we should not modify tx object, but there would be a lot of changes around this. I would suggest keeping it like that and doing research and updating it in one of the next PRs. It would be easier to understand and review.

Reply:
I would prefer not to merge something that we are going to change soon after in a follow-up PR, many things on context can be lost along the way.

Reply:
Anyway, we haven't decided on the solution yet. 
- Gas price is used by several packages.
- This function requires base fee value in order to pre-fill fields.
- This function is just a helper for a tx sender which does not really affect the logic and behavior.  

The option that we have atm is to introduce a new object within each of these packages and do calculations there. It would create several code duplications and affects the architecture since we would have to sync the base fee across all of them and make sure it is in sync every time. I see this option as overengineering (just my opinion).